react/FormConfirmationCheckbox.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { ChangeEvent, useEffect } from 'react'
+import { Checkbox } from 'vtex.styleguide'
+import { IOMessage } from 'vtex.native-types'
+import { useCssHandles } from 'vtex.css-handles'
+
+import {
+  useNewsletterDispatch,
+  useNewsletterState,
+} from './components/NewsletterContext'
+
+interface Props {
+  checkboxLabel?: string
+}
+
+const CSS_HANDLES = [
+  'confirmationCheckboxContainer',
+  'confirmationCheckboxLabel',
+] as const
+
+function FormConfirmationCheckbox(props: Props) {
+  const {
+    checkboxLabel = 'store/newsletter-checkbox-confirmation.checkboxLabel.default',
+  } = props
+
+  const dispatch = useNewsletterDispatch()
+  const { confirmation } = useNewsletterState()
+  const handles = useCssHandles(CSS_HANDLES)
+
+  // Initialize `confirmation` context value to signal that there is a
+  // FormConfirmationCheckbox being rendered inside the newsletter form.
+  useEffect(() => dispatch({ type: 'UPDATE_CONFIRMATION', value: false }), [
+    dispatch,
+  ])
+
+  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
+    dispatch({ type: 'UPDATE_CONFIRMATION', value: e.target.checked })
+  }
+
+  return (
+    <div className={handles.confirmationCheckboxContainer}>
+      <label
```

Comment:
Shouldn't we use the `label`  prop of `Checkbox`?  Also, how would one add links to this label?

Example:

I accept the [Terms of Service](https://google.com) and [Privacy Policy](https://google.com).

Reply:
We could use the label from the component! 

I left it like this for two reasons: 

- enable easier customization, since we can add CSS handles in the label, instead of relying on global selectors;
-  maintain a pattern, since we don't use the built-in `label` props in any of the inputs in the form.

About the links, to be totally honest, I forgot 😅  Will add support for two links, as we talked about :)